test/e2e/snaps/test-snap-update-component.spec.js

```diff
@@ -0,0 +1,163 @@
+          text: 'Connect',
+          tag: 'button',
+        });
+
+        await driver.waitForSelector({ text: 'Install' });
+
+        await driver.clickElementSafe('[data-testid="snap-install-scroll"]');
+
+        await driver.clickElement({
+          text: 'Install',
+          tag: 'button',
+        });
+
+        // wait for permissions popover, click checkboxes and confirm
+        await driver.delay(500);
+        await driver.clickElement('.mm-checkbox__input');
+        await driver.clickElement({
+          text: 'Confirm',
+          tag: 'button',
+        });
+
+        await driver.waitForSelector({ text: 'OK' });
+
+        await driver.clickElement({
+          text: 'OK',
+          tag: 'button',
+        });
+
+        // navigate to test snap page
+        const windowHandles = await driver.waitUntilXWindowHandles(
+          2,
+          1000,
+          10000,
+        );
+        await driver.switchToWindow(windowHandles[1]);
+
+        // wait for npm installation success
+        await driver.waitForSelector({
+          css: '#connectUpdate',
+          text: 'Reconnect to Update Snap',
+        });
+
+        // switch to the original MM tab
+        const extensionPage = windowHandles[0];
+        await driver.switchToWindow(extensionPage);
+        await driver.delay(1000);
+
+        // click on the global action menu
+        await driver.clickElement(
+          '[data-testid="account-options-menu-button"]',
+        );
+
+        // try to click on the snaps item
+        await driver.clickElement({
+          text: 'Snaps',
+          tag: 'div',
+        });
+        await driver.waitForSelector({
+          text: 'BIP-32 Example Snap',
+          tag: 'p',
+        });
+
+        // click into snap view and attempt to update the snap
+        await driver.clickElement({
+          text: 'BIP-32 Example Snap',
+          tag: 'p',
+        });
+        await driver.waitForSelector({
+          css: '.mm-button-link',
+          text: 'Update',
+          tag: 'button',
+        });
+        await driver.clickElement({
+          css: '.mm-button-link',
+          text: 'Update',
+          tag: 'button',
+        });
+
+        await driver.clickElementSafe('[data-testid="snap-update-scroll"]');
+
+        await driver.clickElement({
+          text: 'Update',
+          tag: 'button',
+        });
+
+        await driver.waitForSelector({ text: 'OK' });
+
+        await driver.clickElement({
+          text: 'OK',
+          tag: 'button',
+        });
+        await driver.delay(1000);
+
+        // try to find update link again, succeed if not there
+        // click on the global action menu
+        await driver.clickElement(
+          '[data-testid="account-options-menu-button"]',
+        );
+
+        // try to click on the snaps item
+        await driver.clickElement({
+          text: 'Snaps',
+          tag: 'div',
+        });
+        await driver.delay(1000);
```

Comment:
This `delay` is also good candidate to be replaced.

Reply:
fixed in latest commit

Reply:
This doesn't look like it has been replaced yet